### PR TITLE
ci: add new bash scripts for test artifacts generation

### LINKED ISSUE #899: Adjust test artifacts building process
**Author the user story for this feature**

As a developer, I want to generate all test artifacts to proper directory using a single command.

**Is your feature request related to a problem? Please describe.**
#878

**Describe the solution you'd like**
We need one script for generating all artifacts but we have many artifacts projects, so a proposed solution is to organize scripts in a tree structure where the main script will run platform scripts and platform scripts will run case-specific scripts, so:
* Add new script for generating all artifact android artifacts, reuse existing android scripts, and adjust each if needed.
* Add new script for generating iOS artifacts
* Add main scripts for generating all flank artifacts

**Additional context**
This issue is the next step after #898

### PR DESCRIPTION
**Pull request by jan-gogo**
Fixes #899 

1. Source the `.env` file by running `source .env` in your project root.
2. Run `update_test_artifacts all` and wait until the process will finish with no errors.
3. Check if test artifacts in `test_runner/src/test/kotlin/ftl/fixtures/tmp` are updated correctly.

## Checklist

- [x] Android artifacts
- [x] iOS artifacts
- [x] Go artifacts
- [x] Documented

### CODE REVIEW

File: test_projects/ios/EarlGreyExample/ops.sh [hunk 1]
```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env bash
+
+EARL_GREY_EXAMPLE="$TEST_PROJECTS_IOS/EarlGreyExample"
+
+function setup_ios_env() {
+  if ! [ -x "$(command -v xcpretty)" ]; then
+    gem install cocoapods
```
Comment (axelzuziak-gogo) [inline test_projects/ios/EarlGreyExample/ops.sh:7]: Can we use:
`gem install cocoapods -v 1.9.3`
This gem is getting updated pretty often.

File: test_projects/android/bash/generate_apks.sh [hunk 2]
```diff
@@ -1,5 +1,7 @@
 #!/usr/bin/env bash
 
+echo "DEPRECATED!!!"
```
Comment (axelzuziak-gogo) [inline test_projects/android/bash/generate_apks.sh:3]: Can we add more descriptive message? Like why it was deprecated / what's the new way of generating it etc.

File: test_projects/ios/EarlGreyExample/EarlGreyExample.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -283,18 +283,18 @@
 				TargetAttributes = {
 					2CB7314C1C29E54A00CF35C1 = {
 						CreatedOnToolsVersion = 7.2;
-						DevelopmentTeam = H86XEK7ZTM;
+						DevelopmentTeam = AD2V26JBWL;
```
Comment (axelzuziak-gogo) [inline test_projects/ios/EarlGreyExample/EarlGreyExample.xcodeproj/project.pbxproj:286]: Don't worry about that. We will change it again, once we obtain new Apple developer account

File: test_projects/ios/EarlGreyExample/EarlGreyExample.xcodeproj/xcshareddata/xcschemes/EarlGreyExampleSwiftTests.xcscheme [hunk 4]
```diff
@@ -43,8 +41,6 @@
       debugDocumentVersioning = "YES"
       debugServiceExtension = "internal"
       allowLocationSimulation = "YES">
-      <AdditionalOptions>
-      </AdditionalOptions>
    </LaunchAction>
```
Comment (axelzuziak-gogo) [inline test_projects/ios/EarlGreyExample/EarlGreyExample.xcodeproj/xcshareddata/xcschemes/EarlGreyExampleSwiftTests.xcscheme:44]: This file was edited by Xcode - nothing changed, just XML formatting

File: test_projects/ios/EarlGreyExample/ops.sh [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  fi
+  (cd "$EARL_GREY_EXAMPLE" && pod install)
+}
+
+function install_xcpretty() {
+  if ! [ -x "$(command -v xcpretty)" ]; then
+    gem install xcpretty
```
Comment (axelzuziak-gogo) [inline test_projects/ios/EarlGreyExample/ops.sh:14]: You can specify xcpretty version too, but it's just used for formatting build output so it won't break anything.

**Review by axelzuziak-gogo — COMMENTED** [review]
Some minor comments

File: test_projects/ios/EarlGreyExample/ops.sh (hunk 5 above)
Comment (jan-gogo) [reply test_projects/ios/EarlGreyExample/ops.sh:14]: I will leave it as is

**Review by axelzuziak-gogo — APPROVED** [review]

**Comment by axelzuziak-gogo** [comment]
Looks good 👍

**Review by adamfilipow92 — APPROVED** [review]

**Pull request merged**